koku/api/tags/queries.py

```diff
@@ -41,8 +41,10 @@
 class TagQueryHandler(QueryHandler):
     """Handles tag queries and responses."""
 
+    _DEFAULT_ORDERING = {'tags': 'asc'}
+
     def __init__(self, query_parameters, url_data,
-                 tenant, data_sources, **kwargs):
+                 tenant, default_ordering=None, **kwargs):
```

Comment:
Can we add back `data_sources` as a required parameter?  It's required for this class to do its main job; query the database for tags.

This code would require that the subclass author knows that it needs to set this list after instantiating super.  Sort of a lousy developer experience when they will inevitably forget to set it and need to dig into the parent's source code to understand why it's not working.

Reply:
It's the same thing as DRF's serializers requiring you to set fields attributes on the class. Data sources is really a class attribute, not parameters.

But, there is definitely some tidying up I can do, to make it clearer how sub-classes need to be implemented.

Reply:
DRF's serializers are intentionally designed to be flexible and sub-classed to meet a wide array of use cases. Here we have a very specific class that we maintain and are the sole users of. If the param is required for the class to work, we should have the param. 

Please defer tidying up. Let's fix broken functionality and get this to a place where we can merge it. We have a full sprint on deck and need to move on.

Reply:
The param is not required for the class to work. Thus, the change.

Reply:
without setting `self.data_sorurces` then `get_tags()` will not do anything since it will be walking over an empty list.  To me that seems like it's required. 🤷‍♂

Reply:
It's not required *to be a parameter*. It's a required class attribute.

There is a difference.

The class needs the value, but external consumers of the class are setting that value by passing it in. It's model data defined by sub-classes. You don't need to pass it around from child to parent via __init__().